src/parser.cpp

```diff
@@ -978,19 +978,23 @@ namespace ada::parser {
           break;
         }
         case ada::state::PORT: {
-          auto out = helpers::parse_port(std::string_view(pointer, pointer_end - pointer),
-                                         state,
-                                         url.is_valid,
-                                         url.is_special());
-
-          if (out.has_value()) {
-            if (url.scheme_default_port() == out) {
+          uint16_t port{};
+          auto r = std::from_chars(pointer, pointer_end, port);
```

Comment:
Can ae make this a function so that we don’t need to have the same exact implementation in 2 different places?

Reply:
Done.